File: cap11/cap11-backend/src/main/java/fr/eni/ep3jasp/angular/backend/domain/Cuenta.java
package fr.eni.ep3jasp.angular.backend.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.Type;
import java.time.LocalDateTime;
import java.util.*;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Data
@NoArgsConstructor
@EqualsAndHashCode(exclude={"id"})
@Entity
public class Cuenta {
  @Id
  @GeneratedValue
  private Long id;
  private Integer numero;
  private String nombre;

  public Cuenta(Integer numero, String nombre) {
    this.id=null;
    this.numero=numero;
    this.nombre=nombre;
  }
}

File: cap18/cap18.mvc/src/main/kotlin/fr/eni/ep3jasp/cap05/kotlin/spring5/mvc/app/domain/Cuenta.kt
package fr.eni.ep3jasp.cap05.kotlin.spring5.mvc.app.domain

import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.Type
import java.time.LocalDateTime
import java.util.*
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id



// tag::sbmvckt09[]
@Entity
data class Cuenta(
        @Id
        @GeneratedValue(generator = "uuid")
        @GenericGenerator(name = "uuid", strategy = "uuid2")
        @Type(type = "uuid-char")
        var id: UUID? = null,
        var aperturaDateTime: LocalDateTime = LocalDateTime.now(),
        var nombre: String = ""
)
// end::sbmvckt09[]
